File: my-folder/0001-two-sum/solution.go
func twoSum(nums []int, target int) []int {
    for i := 0; i < len(nums); i++ {
		for j := 0; j < len(nums); j++ {
			if i != j && nums[i]+nums[j] == target {
				slice := []int{i, j}
				return slice
			}
		}
	}

	return nil
}

File: my-folder/0001-two-sum/solution.rs
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut possible_pair_hash: HashMap<i32, i32> = HashMap::new();
        let mut sum_pair: Vec<i32> = Vec::new();

        for i in 0..nums.len() {
            let tmp = target - nums[i];

            if !possible_pair_hash.contains_key(&tmp) {
                possible_pair_hash.insert(nums[i], i as i32);
            } else {
                sum_pair.push(*possible_pair_hash.get(&tmp).unwrap());
                sum_pair.push(i as i32);
                return sum_pair;
            }
        }

        return vec![];
    }
}
